third_party/move/move-core/types/src/transaction_argument.rs

```diff
@@ -90,9 +93,13 @@ impl TryFrom<MoveValue> for TransactionArgument {
 pub fn convert_txn_args(args: &[TransactionArgument]) -> Vec<Vec<u8>> {
     args.iter()
         .map(|arg| {
-            MoveValue::from(arg.clone())
-                .simple_serialize()
-                .expect("transaction arguments must serialize")
+            if let TransactionArgument::Raw(bytes) = arg {
+                bytes.clone()
+            } else {
+                MoveValue::from(arg.clone())
+                    .simple_serialize()
+                    .expect("transaction arguments must serialize")
```

Comment:
Nit: maybe change this to an invariant violation error? For the future....

Reply:
This should come from existing logic and given the simple value types an argument type I think unwrapping is fine?

Reply:
We probably should have some test then in case more variants are added? We recently had an unwrap in txn arguments which stopped being fine because the actual error became possible, and there were no tests. Results seems safer given that it is a user-defined input, but maybe I am paranoid :)